### PR TITLE
Report Generation - PR

### PR DESCRIPTION
**Pull request by Leon-JS02**
Scripts have been added to:
- Use reportlab to build a PDF file from an IO stream
- Send the PDF file as an email attachment through an SES client
- Query the database for useful weekly summary information
- Lambda handler file - runs when the AWS lambda is triggered
- Dockerise the report generator
- Build, tag, and push the Docker image to an ECR repository (automatically created)

(Code refactor, and PDF style changes are very needed).

### CODE REVIEW

File: report/database_handler.py
Comment (mehrdadhalali) [inline report/database_handler.py:1]: Don't forget the readme!

File: report/database_handler.py [hunk 1]
```diff
@@ -25,3 +25,105 @@ def get_connection() -> connection:
...
+    genre_count_dict = {}
+    other_count = 0
+    for i, row in enumerate(results):
+        if i < 8:
+            genre_count_dict[row['genre_name']] = row['genre_count']
+        else:
+            other_count += row['genre_count']
+    if other_count > 0:
+        genre_count_dict['Other'] = other_count
+    return genre_count_dict
```
Comment (mehrdadhalali) [inline report/database_handler.py:62]: This feels like it could be its own function, to make testing easier.

File: report/database_handler.py
Comment (Leon-JS02) [reply report/database_handler.py:1]: On it right now!

File: report/generate_report.py [hunk 2]
```diff
@@ -1 +1,130 @@
...
+from reportlab.platypus import SimpleDocTemplate, Paragraph, Table, TableStyle
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.platypus import Spacer
+
+
+def get_table_style() -> TableStyle:
+    """Returns a table style with a dark theme."""
+    return TableStyle([
+        ('BACKGROUND', (0, 0), (-1, 0), colors.HexColor("#7f5ace")),
+        ('TEXTCOLOR', (0, 0), (-1, 0), colors.whitesmoke),
```
Comment (mehrdadhalali) [inline report/generate_report.py:16]: Nice colour

**Review by mehrdadhalali — APPROVED** [review]
Very nicely done!

**Review by mehrdadhalali — APPROVED** [review]

**Review by paulauus — APPROVED** [review]

**Pull request merged**